File: app/src/main/java/fmrsabino/moviesdb/ui/base/BaseFragment.java
package fmrsabino.moviesdb.ui.base;

import android.support.v4.app.Fragment;

public abstract class BaseFragment extends Fragment {
    @Override
    public void onResume() {
        super.onResume();
        onViewAttached();
    }

    @Override
    public void onPause() {
        super.onPause();
        onViewDetached();
    }

    protected abstract void onViewAttached();
    protected abstract void onViewDetached();
}

File: app/src/main/java/fmrsabino/moviesdb/ui/base/BaseFragment.kt
package fmrsabino.moviesdb.ui.base

import android.support.v4.app.Fragment

abstract class BaseFragment : Fragment() {
    override fun onResume() {
        super.onResume()
        onViewAttached()
    }

    override fun onPause() {
        super.onPause()
        onViewDetached()
    }

    protected abstract fun onViewAttached()
    protected abstract fun onViewDetached()
}
